flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/rules/physical/batch/BatchExecSinkRule.scala

```diff
@@ -47,25 +47,26 @@ class BatchExecSinkRule extends ConverterRule(
     sinkNode.sink match {
       case partitionSink: PartitionableTableSink
         if partitionSink.getPartitionFieldNames != null &&
```

Comment:
According to the API's comment of `PartitionableTableSink`, we don't have to check whether this is null